server/chat-plugins/youtube.ts

```diff
@@ -129,6 +143,27 @@ export class YoutubeInterface {
 		if (!(id in this.data.channels)) return this.getChannelData(id, username);
 		return Promise.resolve({...this.data.channels[id]});
 	}
+	async getVideoData(link: string): Promise<VideoData | null> {
+		const id = this.getId(link);
+		const raw = await Net(`${ROOT}videos`).get({
+			query: {part: 'snippet,statistics', id, key: Config.youtubeKey},
+		});
+		const res = JSON.parse(raw);
+		if (!res || !res.items || res.items.length < 1) return null;
+		const video = res.items[0];
+		return {
+			title: video.snippet.title,
+			id,
+			date: new Date(video.snippet.publishedAt).toString(),
+			description: video.snippet.description,
+			channelTitle: video.snippet.channelTitle,
+			channelUrl: video.snippet.channelId,
+			views: video.statistics.viewCount,
+			thumbnail: video.snippet.thumbnails.default.url,
+			likes: video.statistics.likeCount,
+			dislikes: video.statistics.dislikeCount,
+		} as VideoData;
```

Comment:
Is the `as VideoData` necessary? It can cover up bugs and should probably be handled individually.